### PR TITLE
Js card dialog updates

### PR DESCRIPTION
**Pull request by jon571**
refactors baseline dialogs to to work with new card/dialog functionality.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,28 +1,5 @@
 {
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "editor.formatOnSave": true,
-  "editor.formatOnPaste": false,
```
Comment (Fuhji) [inline .vscode/settings.json:4]: RIP Peacock
image

File: src/app/pages/quickMaps/pages/baselineDetails/baselineDetails.component.html [hunk 2]
```diff
@@ -3,8 +3,10 @@
   <app-base-est></app-base-est>
   <gridster-item [item]="widget" *ngFor="let widget of dashboard">
     <!-- Map View -->
-    <app-map-view [widget]="widget" [resizeEvent]="resizeEvent" *ngIf="widget.type === 'widgetMap'"></app-map-view>
-
+    <app-card2 #card [widget]="widget" [resizeEvent]="resizeEvent" *ngIf="widget.type === 'widgetMap'">
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/baselineDetails/baselineDetails.component.html:6]: when it all refactored are you planning to find+replace to something other than `app-card2`?

File: src/app/pages/quickMaps/pages/baselineDetails/foodItems/foodItems.component.scss [hunk 3]
```diff
@@ -1,42 +1 @@
-@import 'src/assets/scss/_colour';
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/baselineDetails/foodItems/foodItems.component.scss:1]: did you intent to remove all scss except a single commented out line from fooditems?

File: .vscode/settings.json (hunk 1 above)
Comment (jon571) [reply .vscode/settings.json:4]: :-)

File: src/app/pages/quickMaps/pages/baselineDetails/map-view/map-view.component.scss [hunk 4]
```diff
@@ -1,14 +1 @@
...
-  & ::ng-deep {
-    & .mat-tab-body-wrapper {
-      flex: 1 1;
-      & .mat-tab-body-content > * {
-        height: 100%;
-      }
-    }
-  }
-}
+// @import 'src/assets/scss/_colour.scss';
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/baselineDetails/map-view/map-view.component.scss:1]: are component scss files required any more if they are not used?

File: src/app/pages/quickMaps/pages/baselineDetails/baselineDetails.component.html (hunk 2 above)
Comment (jon571) [reply src/app/pages/quickMaps/pages/baselineDetails/baselineDetails.component.html:6]: I'll actually move the code from card2 to card to maintain the git history. :-)

File: src/app/pages/quickMaps/pages/baselineDetails/monthlyFood/monthlyFood.component.ts [hunk 5]
```diff
@@ -53,24 +57,36 @@ export class MonthlyCardComponent implements OnInit, OnDestroy {
...
     private currentDataService: CurrentDataService,
     private quickMapsService: QuickMapsService,
     private dialogService: DialogService,
     private cdr: ChangeDetectorRef,
-  ) {}
+    @Optional() @Inject(MAT_DIALOG_DATA) public data?: DialogData,
+  ) { }
 
   ngOnInit(): void {
-    this.resizeSub = this.resizeEvent.subscribe((widget) => {
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/baselineDetails/monthlyFood/monthlyFood.component.ts:64]: I might have missed this, but how are resize events managed for individual components now?

**Review by Fuhji — COMMENTED** [review]
Added some comments which may have been answered elsewhere.

File: src/app/pages/quickMaps/pages/baselineDetails/foodItems/foodItems.component.scss (hunk 3 above)
Comment (jon571) [reply src/app/pages/quickMaps/pages/baselineDetails/foodItems/foodItems.component.scss:1]: Yeah.  I include a second default scss file in all those cards now so this one is only for component specific scss, of which there is none.

I like to leave the colors there to prompt its use! Although I often comment it out.

File: src/app/pages/quickMaps/pages/baselineDetails/map-view/map-view.component.scss (hunk 4 above)
Comment (jon571) [reply src/app/pages/quickMaps/pages/baselineDetails/map-view/map-view.component.scss:1]: No content but better left in

File: src/app/pages/quickMaps/pages/baselineDetails/monthlyFood/monthlyFood.component.ts (hunk 5 above)
Comment (jon571) [reply src/app/pages/quickMaps/pages/baselineDetails/monthlyFood/monthlyFood.component.ts:64]: You subscribe to a observable on the card e.g.:
https://github.com/micronutrientsupport/micronutrient-support-tool/blob/js_card_dialog_updates/src/app/pages/quickMaps/pages/baselineDetails/map-view/map-view.component.ts#L75

**Comment by jon571** [comment]
Answered comments :-)

**Review by Fuhji — APPROVED** [review]

**Pull request merged**